src/formatDate.js

```diff
@@ -48,9 +48,46 @@
  *
  * @returns {string}
  */
-
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const newDate = [];
+  const oldSeparator = fromFormat[3];
+  const newSeparator = toFormat[3];
+  const divideDate = date.split(oldSeparator);
+  const dateMap = {};
+
+  for (let i = 0; i < divideDate.length; i++) {
+    dateMap[fromFormat[i]] = divideDate[i];
+  }
+
+  if (!dateMap['YY']) {
+    dateMap['YY'] = dateMap['YYYY'].slice(-2);
+  }
+
+  if (!dateMap['YYYY']) {
+    dateMap['YYYY'] = +dateMap['YY'] >= 30 ? `19${dateMap['YY']}` : `20${dateMap['YY']}`;
```

Comment:
Looks better, agree?
```suggestion
    dateMap['YYYY'] = +dateMap['YY'] >= 30
      ? `19${dateMap['YY']}` 
      : `20${dateMap['YY']}`;
```